src/components/Contacts/ContactDetails/ContactDontationsTab/PartnershipInfo/EditPartnershipInfoModal/EditPartnershipInfoModal.tsx

```diff
@@ -0,0 +1,431 @@
+            <DialogContent dividers>
+              <ContactInputWrapper>
+                <FormControl fullWidth>
+                  <InputLabel id="status-select-label">
+                    {t('Status')}
+                  </InputLabel>
+                  <Select
+                    labelId="status-select-label"
+                    value={status}
+                    onChange={(e) =>
+                      updateStatus(e.target.value as StatusEnum, setFieldValue)
+                    }
+                  >
+                    {Object.values(StatusEnum).map((value) => (
+                      <MenuItem key={value} value={value}>
+                        {t(value)}
+                      </MenuItem>
+                    ))}
+                  </Select>
+                </FormControl>
+              </ContactInputWrapper>
+              <ContactInputWrapper>
+                <FormControl fullWidth>
+                  <InputLabel>{t('Amount')}</InputLabel>
+                  <Input
+                    value={pledgeAmount}
+                    type="number"
+                    disabled={status !== StatusEnum.PartnerFinancial}
+                    aria-readonly={status !== StatusEnum.PartnerFinancial}
+                    onChange={handleChange('pledgeAmount')}
+                    inputProps={{ 'aria-label': t('Amount') }}
+                    endAdornment={
+                      <Tooltip
+                        title={
+                          <Typography>
+                            {t(
+                              'Commitments can only be set if status is Partner - Financial',
+                            )}
+                          </Typography>
+                        }
+                      >
+                        <InfoIcon />
+                      </Tooltip>
+                    }
+                    fullWidth
+                  />
+                </FormControl>
+              </ContactInputWrapper>
+              <ContactInputWrapper>
+                <FormControl fullWidth>
+                  <InputLabel id="currency-select-label">
+                    {t('Currency')}
+                  </InputLabel>
+                  <Select
+                    labelId="currency-select-label"
+                    value={pledgeCurrency}
+                    onChange={handleChange('pledgeCurrency')}
+                    MenuProps={{
+                      anchorOrigin: {
+                        vertical: 'bottom',
+                        horizontal: 'left',
+                      },
+                      transformOrigin: {
+                        vertical: 'top',
+                        horizontal: 'left',
+                      },
+                      getContentAnchorEl: null,
+                      PaperProps: {
+                        style: {
+                          maxHeight: '300px',
+                          overflow: 'auto',
+                        },
+                      },
+                    }}
```

Comment:
I found this fix for the select menus being positioned strange, especially for larger lists. So now it's only ```300px``` and is directly/correctly positioned under the select input.